@claude fix error

Adding thiserror-impl v1.0.69
      Adding tokio-cron-scheduler v0.10.2 (available: v0.14.0)
      Adding uuid v1.17.0
      Adding windows-core v0.61.0 (available: v0.61.2)
      Adding windows-implement v0.60.0
      Adding windows-interface v0.59.1
      Adding windows-strings v0.4.2
 Downloading crates ...
  Downloaded iana-time-zone v0.1.63
  Downloaded num-traits v0.2.19
  Downloaded uuid v1.17.0
  Downloaded num-derive v0.3.3
  Downloaded cron v0.12.1
  Downloaded nom v7.1.3
  Downloaded chrono v0.4.41
  Downloaded tokio-cron-scheduler v0.10.2
  Downloaded minimal-lexical v0.2.1
  Downloaded syn v1.0.109
warning: scheduler v0.1.0 (/opt/render/project/src/crates/scheduler) ignoring invalid dependency `orchestrator` which is missing a lib target
   Compiling common v0.1.0 (/opt/render/project/src/crates/common)
   Compiling num-traits v0.2.19
error[E0521]: borrowed data escapes outside of method
  --> crates/common/src/crawler.rs:42:13
   |
34 |       pub async fn run_all(&self) -> CrawlerResult<()> {
   |                            -----
   |                            |
   |                            `self` is a reference that is only valid in the method body
   |                            let's call the lifetime of this reference `'1`
...
42 | /             tasks.spawn(async move {
43 | |                 match crawler.run().await {
44 | |                     Ok(_) => {
45 | |                         info!("{} completed successfully", name);
...  |
53 | |             });
   | |              ^
   | |              |
   | |______________`self` escapes the method body here
   |                argument requires that `'1` must outlive `'static`
For more information about this error, try `rustc --explain E0521`.
error: could not compile `common` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
==> Build failed 😞
==> Common ways to troubleshoot your deploy: https://render.com/docs/troubleshooting-deploys
Need better ways to work with logs? Try theRender CLIor set up a log stream integration

--- a/crates/common/Cargo.toml
+++ b/crates/common/Cargo.toml
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 anyhow = "1.0"
 async-trait = "0.1"
+futures = "0.3"
 reqwest = { version = "0.12", features = ["json"] }
 thiserror = "1.0"
 tokio = { version = "1.0", features = ["rt", "rt-multi-thread", "macros"] }

--- a/crates/common/src/crawler.rs
+++ b/crates/common/src/crawler.rs
@@ -32,14 +32,12 @@ impl CrawlerManager {
     }
 
     pub async fn run_all(&self) -> CrawlerResult<()> {
-        use tokio::task::JoinSet;
+        use futures::future::join_all;
         use tracing::{info, warn};
 
-        let mut tasks = JoinSet::new();
-        
-        for crawler in &self.crawlers {
+        let futures = self.crawlers.iter().map(|crawler| {
             let name = crawler.name();
-            tasks.spawn(async move {
+            async move {
                 match crawler.run().await {
                     Ok(_) => {
                         info!("{} completed successfully", name);
@@ -50,16 +48,18 @@ impl CrawlerManager {
                         Err(e)
                     }
                 }
-            });
-        }
+            }
+        });
 
+        let results = join_all(futures).await;
+        
         let mut success_count = 0;
         let mut error_count = 0;
 
-        while let Some(result) = tasks.join_next().await {
+        for result in results {
             match result {
-                Ok(Ok(_)) => success_count += 1,
-                Ok(Err(_)) | Err(_) => error_count += 1,
+                Ok(_) => success_count += 1,
+                Err(_) => error_count += 1,
             }
         }
 

